addons/website/models/res_users.py

```diff
@@ -26,6 +27,19 @@ def _has_unsplash_key_rights(self):
             return True
         return super(ResUsers, self)._has_unsplash_key_rights()
 
+    @api.constrains('login', 'website_id')
+    def _check_login(self):
+        users_without_website = self.filtered(lambda u: not u.website_id)
+        if users_without_website:
+            self.env.cr.execute(""" SELECT COUNT(*)
+                FROM res_users
+                WHERE login IN %s
+                    AND website_id IS NULL
+            """, (tuple(users_without_website.mapped('login')),))
```

Comment:
* Should flush before executing any SQL query.
* I'm not a great fan of filtered
* What about this single sql query:
```
SELECT login, count(*) as nbr
  FROM res_users 
 WHERE login in (select login from res_users where id in %s AND website_id IS NULL)
   AND website_id IS NULL
HAVING nbr>1
```

Reply:
The group by is missing (at least) in my query

Reply:
@nseinlet Thanks for the review!
The commit has been updated accordingly.